### PR TITLE
RSE-761: testdeck replacement

### PR DESCRIPTION
**Pull request by ronaveva**
First run to replace testdeck suite with test containers and spock based tests

**Describe the solution you've implemented**
* Use testcontainers to run rundeck server image
* basic API client setup via OkHttp
* Migrate some API tests from old bash scripts to Spock
* Migrate selenium tests to spock

**Additional context**
Tests can be run using testcontainers to automatically start rundeck server in a docker container, or can be executed against any URL using environment parameters, see README file

**TODO**
- [x] run API tests against docker image or running instance 
- [x] migrate selenium tests
- [x] change circleci configuration new test module instead of testdeck
- [x] improve test logging

### CODE REVIEW

File: .circleci/config.yml
Comment (gschueler) [inline .circleci/config.yml:494]: I think the tomcat tag is not set correctly
```suggestion
            TOMCAT_TAG:<< parameters.tomcat-vers >>
```

File: functional-test/src/test/groovy/org/rundeck/util/container/RdDockerContainer.groovy
Comment (gschueler) [inline functional-test/src/test/groovy/org/rundeck/util/container/RdDockerContainer.groovy:40]: may want to use a log consumer
```
        withLogConsumer(new Slf4jLogConsumer(log))
```

File: .circleci/config.yml
Comment (gschueler) [inline .circleci/config.yml:494]: should use an env var that makes sense, it is not a rundeck image, it is a tomcat tag. I think it makes more sense as TOMCAT_TAG

Comment (ronaveva) [reply .circleci/config.yml]: i think, instead of an specific name we should only use a generic one, this way we don't have multiple variables with the same purpose

File: functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/BasicLoginSpec.groovy
Comment (gschueler) [inline functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/BasicLoginSpec.groovy:22]: why was this removed?

**Comment by gschueler** [comment]
there is a fix for CI packaging tests in main, will need to update

File: functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/BasicLoginSpec.groovy
Comment (ronaveva) [reply functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/BasicLoginSpec.groovy:22]: it's better to validate against elements since pageSource gets evaluated before all of the javascript/jquery/ajax, and it wasn't working

Comment (gschueler) [reply functional-test/src/test/groovy/org/rundeck/tests/functional/selenium/BasicLoginSpec.groovy:22]: ah, ok.  I was trying to replicate the original test here https://github.com/rundeck/rundeck/blob/main/test/selenium/src/__tests__/selenium/login.test.ts#L31 but I was actually not doing it correctly anyway.  I will update it

**Review by gschueler — APPROVED** [review]
will need to wait until #8610 is merged

**Review by gschueler — CHANGES_REQUESTED** [review]
Circle config needs to be updated to match new design

**Review by gschueler — APPROVED** [review]

**Pull request merged**
